### ISSUE TITLE
lower-case time formats?

### ISSUE THREAD

**Issue by sql-sith** [JanDeDobbeleer/oh-my-posh#532]
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] I looked for duplicate issues before submitting this one

### Description

This is a feature/wish issue.

Call me timid, but I really like lower-case letters in my prompt. I don't know a way to do that using `golang`'s "magic date" formatting. Is it possible to do this with the `golang` format? If not, should I do this with environment variables in oh-my-posh? I already do this to get command duration, using `bash-preexec.sh` so I already have the infrastructure in place to do it with environment variables if needed.

### Environment
- Oh my Posh version: 3.68.4 or 3.68.5, depending on which environment it is
- Theme: custom
- Operating System: Windows 10 and WSL, both versions 1 and 2, running Ubuntu 18.04 and 20.04
- Shell:  Powershell 7.1.3, bash 5.0.17(1)-release (on Ubuntu 20.4), and bash 4.4.20(1)-release (on Ubuntu 18.4)
- Terminal: Windows Terminal 1.7.572.0

### Steps to Reproduce

1. Use the time segment
2. Include lower-case tokens such as jan or mon. These are treated like literals and not expanded
3. OR include upper-case tokens such as Jan or Mon. These are expanded as proper tokens but I cannot force them to be lower-case

**Expected behavior:** lower-case expansion of tokens ... somehow

**Actual behavior:** either no expansion or upper-case expansion, depending on which repro you do from the list above.

I'll include a screenshot of my these, just because I'm proud of it. Without this project, I'd never have a prompt this nice!

![image](https://user-images.githubusercontent.com/1082694/111179788-657b0f00-857a-11eb-8364-ede89127c03a.png)

**Comment by sql-sith** [comment]
<img width="988" alt="omp3-mockup" src="https://user-images.githubusercontent.com/1082694/111178474-31531e80-8579-11eb-8765-a849f2b2fdbc.png">

**Comment by sql-sith** [comment]
_needs to change the os logo for windows :)_

**Comment by TravisTX** [comment]
Something like this should do it.  Just use lowercase letters in your expression:

```json
"time_format": "mon jan _2 15:04:05 2006"
```

**Comment by tradiff** [comment]
I agree with your assessment. I don't think this is possible without changes to OMP.

Regarding a change to OMP, I won't have time to put into this for a few days. Maybe someone else can.  Perhaps we could leverage sprig templates (which has a `lower` function).

**Comment by tradiff** [comment]
Regarding environment variables, if you can set an environment variable to the string you want to display, you can definitely use the [Environment Variable](https://ohmyposh.dev/docs/environment) segment to display that string.

**Comment by JanDeDobbeleer** [comment]
> Perhaps we could leverage sprig templates (which has a lower function)

@TravisTX good idea. Let's add a template to this segment so we can work with that. I'll need a few days though. Busy with changing roles at my current company 😅

**Comment by sql-sith** [comment]
Thanks to you both!

**Comment by JanDeDobbeleer** [comment]
@lnu can you give a template sample here for @sql-sith so we can close this?

**Comment by lnu** [comment]
This should work with the latest version(3.114.0):
```
  {
          "type":"time",
          "style":"plain",
          "properties": {
            "template": "{{.CurrentDate | date \"Mon 2-Jan-2006 15:04:05\" | lower}}"
          }
  },
```
=>
![image](https://user-images.githubusercontent.com/1829553/111433167-3beaf080-86fe-11eb-9f10-721b03d9b479.png)

**Comment by sql-sith** [comment]
Thank you! That was fast. :)

This worked perfectly, right away, on my *nix shells. However, I'm seeing behavior I don't expect on Powershell. When I plug in the template above, only the time is displaying. And when I run `oh-my-posh --version`, I am getting back `3.68.4` but `Get-Module oh-my-posh -ListAvailable` shows `3.115.0`. I completely uninstalled/removed and then resinstalled the module and the behavior is the same. I also checked to make sure I didn't have any installations from `choco` or `scoop` loitering around. Does this behavior sound like any issues someone has had in the past? 'Cuz I'm stumped so far.

**Comment by lnu** [comment]
what does your powershell profile look like?

how do you init the prompt?

**Comment by JanDeDobbeleer** [comment]
@sql-sith the `oh-my-posh` command implies you have it installed using scoop (or homebrew on macOS). That one isn't shipped like that inside the Powershell module which does the "heavy" lifting for you. If you use something like` Invoke-Expression (oh-my-posh --init --shell pwsh --config mytheme.omp.json)` inside your `$PROFILE`, you should be all set. If you actually use the module and `Set-PoshPrompt -Theme mytheme.omp.json` you need to update oh-my-posh using `Update-Module oh-my-posh -Scope CurrentUser` and restart you Powershell terminal (or create a new tab).

**Comment by sql-sith** [comment]
Thanks for the info @JanDeDobbeleer . I did some digging and I had both the module and oh-my-posh.exe installed. I removed the alias to oh-my-posh.exe. Then I imported the OMP module and ran the `Set-PoshPrompt` cmdlet you gave above and everything's working as expected.

**Comment by sql-sith** [comment]
@lnu , thanks for the reply. Your questions and Jan's info made me take a look at my `$PROFILE` code for OMP again. I had always wondered why a) I had so much code in there, and b) why `executiontime` didn't work for me. Even though I did not have `scoop` installed currently, I know I did in the past. So my inference is that I probably installed using `scoop`, added a lot of code to my profile to customize things (such as calculating my own `executiontime` environment variable, and then at some point I uninstalled `scoop`. Switching over to the module made the config so much simpler.

I did have to fiddle with whitespace a little bit to get my two `right` prompts to right-align. For `time`, I am using simple one-byte diamond characters: `"leading_diamond": "\uE0B6"` and `"trailing_diamond": "\uE0B4". For `executiontime`, which displays on the next line, I am using `"leading_diamond": "<#555555>[ elapsed: </>"` and `"trailing_diamond": "<#555555>]  </>"`. Note the trailing spaces for `executiontime`. Not a big deal at all, and easy to deal with on our side, but I thought you might want to know.

I am going to file another related question in a new issue so that this one can rest in peace. Thank you both!
